discovery/discovery.go

```diff
@@ -85,8 +86,11 @@ func (o *orchestratorPool) GetOrchestrators(numOrchestrators int, suspender comm
 			infoCh <- info
 			return
 		}
-		if err != nil && monitor.Enabled {
-			monitor.LogDiscoveryError(err.Error())
+		if err != nil && !errors.Is(err, context.Canceled) && !errors.Is(err, context.DeadlineExceeded) {
```

Comment:
I think we should still log an error if the error wraps a `context.DeadlineExceeded` as that would indicate a timeout with the O which still feels useful to be aware of as it tells us the O could not respond within the timeout while a `context.Cancelled` (and empirically, the majority of excessive logging on mainnet for Bs seems to be from `context.Cancelled`) is always B initiated and doesn't really tell us anything about O's ability to respond. WDYT?

Reply:
Fixed and rebased